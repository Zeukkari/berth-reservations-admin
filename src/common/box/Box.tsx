import React from 'react';
import classNames from 'classnames';

import styles from './box.module.scss';

export type Props = {
  children: React.ReactNode;
  align?: 'start' | 'center' | 'end' | 'baseline' | 'stretch';
  alignContent?: 'start' | 'center' | 'end' | 'between' | 'around' | 'stretch';
  alignSelf?: 'start' | 'center' | 'end' | 'stretch';
  basis?: 'auto' | 'full' | '1/2' | '1/3' | '2/3' | '1/4' | '2/4' | '3/4';
  justify?:
    | 'around'
    | 'between'
    | 'center'
    | 'end'
    | 'evenly'
    | 'start'
    | 'stretch';
  flexDirection?: 'column' | 'row';
  flex?: '1' | '0';
} & React.DOMAttributes<HTMLButtonElement>;

const Box: React.SFC<Props> = ({
  children,
  align = 'start',
  alignContent = 'contained',
  alignSelf = 'standard',
  basis = 'auto',
  flexDirection = 'column',
  justify,
  flex = 1,
}) => (
  <div className={classNames(styles.container)}>
    <div
      data-align={align}
      data-aligncontent={alignContent}
      data-alignself={alignSelf}
      data-basis={basis}
      data-flex-direction={flexDirection}
      data-justify-content={justify}
      className={classNames(styles.box, styles.flex, styles.basis)}
    >
      {children}
    </div>
  </div>
);

export default Box;
