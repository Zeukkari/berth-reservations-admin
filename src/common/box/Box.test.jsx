import React from 'react';
import { shallow } from 'enzyme';

import Box from './Box';

describe('Box', () => {
  const getWrapper = () => shallow(<Box>Content</Box>);

  it('renders normally', () => {
    const wrapper = getWrapper();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
