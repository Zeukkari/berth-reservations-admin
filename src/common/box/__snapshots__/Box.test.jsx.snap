// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders normally 1`] = `"<div class=\\"container\\"><div data-align=\\"start\\" data-aligncontent=\\"contained\\" data-alignself=\\"standard\\" data-basis=\\"auto\\" data-flex-direction=\\"column\\" class=\\"box flex basis\\">Content</div></div>"`;
